tests/zfs-tests/tests/functional/failmode/failmode_dmu_tx_wait.ksh

```diff
@@ -0,0 +1,94 @@
+#!/bin/ksh -p
+# SPDX-License-Identifier: CDDL-1.0
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or https://opensource.org/licenses/CDDL-1.0.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2025, Klara, Inc.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+log_assert "dmu_tx_assign() blocks when pool suspends with failmode=wait"
+
+function cleanup
+{
+	zinject -c all || true
+	zpool clear $TESTPOOL || true
+	destroy_pool $TESTPOOL
+}
+
+log_onexit cleanup
+
+DISK=${DISKS%% *}
+
+# create a single-disk pool, set failmode=wait
+log_must zpool create -o failmode=wait -f $TESTPOOL $DISK
+log_must zfs create -o recordsize=128k $TESTPOOL/$TESTFS
+
+# start writing to a file in the background. these args to dd will make it
+# keep writing until it fills the pool, but we will kill it before that happens.
+dd if=/dev/urandom of=/$TESTPOOL/$TESTFS/file bs=128k &
+typeset -i dd_pid=$!
+
+# give it a moment for the write throttle to start pushing back
+sleep 2
+
+# force the pool to suspend by inducing the writes and followup probe to fail
+zinject -d $DISK -e io -T write $TESTPOOL
+zinject -d $DISK -e nxio -T probe $TESTPOOL
+
+# should only take a moment, but give it a chance
+log_note "waiting for pool to suspend"
+typeset -i tries=10
+until [[ $(kstat_pool $TESTPOOL state) == "SUSPENDED" ]] ; do
+	if ((tries-- == 0)); then
+		log_fail "pool didn't suspend"
```

Comment:
Do we need to kill the dd pid if we log_fail? Feels like it might cause issues with the cleanup function

Reply:
It's anyone's guess whether or not its blocked in the kernel at this point, but we can try! Last push kills it in cleanup if it's still there.